src/proposals/20230913/SparkEthereum_20231129.t.sol

```diff
@@ -0,0 +1,179 @@
+
+        // Take snapshot before simulating protocol activity
+        uint256 snapshot = vm.snapshot();
+
+        // Simulate 100 days of protocol activity
+        for (uint256 i = 0; i < 2400; i++) {
+            vm.warp(startingTimestamp + i * 1 hours);
+            _supply(DAI, 1e18);
+        }
+
+        // Demonstrate that the asset liability diff would continue to increase over time
+        assertEq(_getAssetLiabilityDiff(DAI),             771_448.289653373654814223 ether);
+        assertEq(_getAssetLiabilityDiff(DAI) - startDiff, 555_263.938288374126089205 ether);
+
+        // Warp back to original timestamp and snapshot
+        vm.warp(startingTimestamp);
+        vm.revertTo(snapshot);
+
+        // Execute spell
+        GovHelpers.executePayload(vm, payload, executor);
+
+        ( uint256 supplyRate1, uint256 variableBorrowRate1 ) = _getRates(DAI);
+
+        // Diff should be small since utilization is almost 100%
+        assertEq(supplyRate1,         0.048418992886337042226176019e27);
+        assertEq(variableBorrowRate1, 0.053790164207174267760128000e27);
+
+        // Supply a small amount of DAI to update rates
+        _supply(DAI, 1e18);
+
+        ( uint256 supplyRate2, uint256 variableBorrowRate2 ) = _getRates(DAI);
+
+        // Show that rates have updated
+        // Diff is small since utilization is almost 100%
+        assertEq(supplyRate2,         0.053380969243632827980092586e27);
+        assertEq(variableBorrowRate2, 0.053790164207174267760128000e27);
+
+        // Supply rate change is slightly less than supplySpread update because of utilization
+        assertEq(variableBorrowRate2 - variableBorrowRate1, 0);
+        assertEq(supplyRate2 - supplyRate1,                 0.004961976357295785753916567e27);
+
+        // Make sure that diff is the same as start of test
+        assertApproxEqAbs(_getAssetLiabilityDiff(DAI), startDiff, 1);
+
+        // Simulate 100 days of protocol activity
+        for(uint256 i = 0; i < 2400; i++) {
+            vm.warp(startingTimestamp + i * 1 hours);
+            _supply(DAI, 1e18);
+        }
+
+        // Demonstrate that the asset liability diff would continue to increase over time
+        assertEq(_getAssetLiabilityDiff(DAI),             219_386.337857977243480099 ether);
+        assertEq(_getAssetLiabilityDiff(DAI) - startDiff,   3_201.986492977714755081 ether);
+    }
+
+    function _getAssetLiabilityDiff(address asset) internal view returns (uint256 diff) {
+        ( , uint256 accruedToTreasuryScaled,,,,,,,,,, ) = dataProvider.getReserveData(asset);
+
+        ( address aToken,, ) = dataProvider.getReserveTokensAddresses(asset);
+
+        uint256 totalDebt      = dataProvider.getTotalDebt(asset);
+        uint256 totalLiquidity = IERC20(asset).balanceOf(aToken);
+        uint256 totalDeposits  = IERC20(aToken).totalSupply();
+
+        uint256 assets      = totalLiquidity + totalDebt;
+        uint256 liabilities = totalDeposits + accruedToTreasuryScaled;
```

Comment:
This should be `(IERC20(aToken).scaledTotalSupply() + accruedToTreasuryScaled) * liquidityIndex` although for DAI there is no reserve factor so treasury amount will be 0 anyways.

Reply:
Updated